sympy/core/basic.py

```diff
@@ -1587,6 +1587,13 @@ def _eval_rewrite(self, pattern, rule, **hints):
         if pattern is None or isinstance(self, pattern):
             if hasattr(self, rule):
                 hints.pop('deep', None)
+                hints.pop('log', None)
+                hints.pop('basic', None)
+                hints.pop('mul', None)
+                hints.pop('power_base', None)
+                hints.pop('power_exp', None)
+                hints.pop('func', None)
+                hints.pop('multinomial', None)
```

Comment:
It is probably better to add `**kw` to `_eval_rewrite_as_expint` instead of popping all the these keywords. `deep` is an exception as it is used internally by `rewrite`.